pennylane_lightning/lightning_kokkos.py

```diff
@@ -0,0 +1,913 @@
+                device_wires (Wires): wires that get initialized in the state
+            """
+
+            if isinstance(state, self._kokkos_state.__class__):
+                state_data = np.zeros(state.size, dtype=self.C_DTYPE)
+                state_data = self._asarray(state_data, dtype=self.C_DTYPE)
+                state.DeviceToHost(state_data.ravel(order="C"))
+                state = state_data
+
+            # translate to wire labels used by device
+            device_wires = self.map_wires(device_wires)
+
+            state = self._asarray(state, dtype=self.C_DTYPE)
+            output_shape = [2] * self.num_wires
+
+            if len(device_wires) == self.num_wires and Wires(sorted(device_wires)) == device_wires:
+                # Initialize the entire device state with the input state
+                self.syncH2D(self._reshape(state, output_shape))
+                return
+
+            # generate basis states on subset of qubits via the cartesian product
+            basis_states = np.array(list(product([0, 1], repeat=len(device_wires))))
+
+            # get basis states to alter on full set of qubits
+            unravelled_indices = np.zeros((2 ** len(device_wires), self.num_wires), dtype=int)
+            unravelled_indices[:, device_wires] = basis_states
+
+            # get indices for which the state is changed to input state vector elements
+            ravelled_indices = np.ravel_multi_index(unravelled_indices.T, [2] * self.num_wires)
+
+            self._kokkos_state.setStateVector(ravelled_indices, state)  # this operation on device
+
+        def _apply_basis_state(self, state, wires):
+            """Initialize the state vector in a specified computational basis state.
+
+            Args:
+                state (array[int]): computational basis state of shape ``(wires,)``
+                    consisting of 0s and 1s.
+                wires (Wires): wires that the provided computational state should be initialized on
+
+            Note: This function does not support broadcasted inputs yet.
+            """
+            # translate to wire labels used by device
+            device_wires = self.map_wires(wires)
+
+            # length of basis state parameter
+            n_basis_state = len(state)
+
+            if not set(state.tolist()).issubset({0, 1}):
+                raise ValueError("BasisState parameter must consist of 0 or 1 integers.")
+
+            if n_basis_state != len(device_wires):
+                raise ValueError("BasisState parameter and wires must be of equal length.")
+
+            # get computational basis state number
+            basis_states = 2 ** (self.num_wires - 1 - np.array(device_wires))
+            basis_states = qml.math.convert_like(basis_states, state)
+            num = int(qml.math.dot(state, basis_states))
+
+            self._create_basis_state(num)
+
+        def apply_lightning(self, operations, apply_pre_rotated_state=False):
```

Comment:
Hi @vincentmr. I'm working on the documentation, and as I said before it helps a lot (as you did) to have functions with the same intent having the same name, in all backends. But, this is only half of what is needed. We also need to have the same method signature. This included the same input variables, and in `LightningQubit` we have `apply_lightning(self, state, operations)`. Could you, please, go through all the same name functions, and change the signature where it is sensible? It is OK, if not possible, but the best alternative is to do that.

Reply:
It is also acceptable to change the LightningQubit signature, but we need to keep LightningGPU on mind, when doing that.